cookies/views/resource.py

```diff
@@ -535,6 +545,64 @@ def edit_resource_details(request, resource_id):
     return render(request, template, context)
 
 
+@auth.authorization_required(ResourceAuthorization.EDIT, _get_resource_by_id)
+def define_resource_content_region(request, resource_id):
+    """
+    Display the resource with the given id
+    """
+    __isPartOf__ = Field.objects.get(uri='http://purl.org/dc/terms/isPartOf')
+    resource = _get_resource_by_id(request, resource_id)
+    part_relations = resource.relations_to.filter(predicate=__isPartOf__).order_by('sort_order').all()
+    part_relations = [p for p in part_relations if not isinstance(p.source, ContentRegion)]
+
+    context = {
+        'resource': resource,
+        'request': request,
+    }
+    on_valid = request.GET.get('next', None)
+    if on_valid:
+        context.update({'next': on_valid})
+
+    if request.method == 'GET':
+        form = UserDefineContentRegionForm(part_relations=part_relations)
+
+    elif request.method == 'POST':
+        form = UserDefineContentRegionForm(request.POST, part_relations=part_relations)
+        if form.is_valid():
+            name = form.cleaned_data.get('name')
+            content_region_start_resource = form.cleaned_data.get('content_region_start_resource')
+            content_region_start_position = form.cleaned_data.get('content_region_start_position')
+            content_region_end_resource = form.cleaned_data.get('content_region_end_resource')
+            content_region_end_position = form.cleaned_data.get('content_region_end_position')
+            __document__ = Type.objects.get(uri='http://xmlns.com/foaf/0.1/Document')
+            content_region = ContentRegion.objects.create(**{
+                'name': name,
+                'created_by_id': request.user.id,
+                'container': resource.container,
+                'start_position': content_region_start_position,
+                'start_resource': _get_resource_by_id(request, content_region_start_resource),
+                'end_position': content_region_end_position,
+                'end_resource': _get_resource_by_id(request, content_region_end_resource),
+            })
+            content_region.save()
+            __part__ = Field.objects.get(uri='http://purl.org/dc/terms/isPartOf')
```

Comment:
I would not use this relation, since it does not refer to another resource. Let's use extent here, see http://dublincore.org/documents/dcmi-terms/#terms-extent